framework/src/parser/MooseSyntax.C

```diff
@@ -99,6 +99,7 @@ associateSyntax(Syntax & syntax, ActionFactory & action_factory)
 
   registerSyntax("AddOutputAction", "Outputs/*");
```

Comment:
I'm not sure I like the new macro. The problem is that there are two similarly named functions (macros) now with very different 3rd arguments. I'd be fine with the new one if it were possible to get rid of the old one but that's not the case.  So now we have this:

```
registerSyntaxTask("AddKernelAction", "Kernels/*", "add_kernel");
registerSyntaxWithType("AddFunctionAction", "Functions/*", "FunctionName");
```

I never had a problem with making several calls to get everything registered where we allowed multiple "things". I can be convinced but I don't think this is any more clear.